react-app/src/components/search/Results.js

```diff
@@ -53,26 +66,46 @@ const useQuery = () => {
 }
 
 const Results = () => {
+    const categories = useSelector(state => state.categories.categories);
 
     const [isLoaded, setIsLoaded] = useState(false);
+    const [categoryId, setCategoryId] = useState(undefined);
 
     let query = useQuery();
-    const key = query.get("key");
+    const searchKey = query.get("key");
 
     const dispatch = useDispatch()
 
     useEffect(() => {
-        dispatch(getItems(key)).then(() => setIsLoaded(true));
-    }, [dispatch, key])
+        setCategoryId(query.get("category") ? query.get("category") : undefined);
```

Comment:
This should not be needed. If you want to set a default value for state when the component is first created you can pass the value directly into the state's initialization. Shift the `query` initialization above your `categoryId` state initialization and then you can pass `query.get('category') || ''` into the `useState` for `categoryId`

As a note, controlled inputs cannot have an undefined state. Perhaps (as I suggest above) you can set it to `''` when there is no category chosen?

Reply:
@isaacsungpak I see this was just changed in https://github.com/justinrusso/loot-locker/pull/68/commits/a3468c738c8a5cb81fe973f56efdb76312006033 before I posted this. Take note of my mention of the value of the controlled input though. That still applies.